core/templates/pages/topics-and-skills-dashboard-page/modals/unassign-skill-from-topics-modal.component.ts

```diff
@@ -60,14 +61,40 @@ export class UnassignSkillFromTopicsModalComponent
       .fetchTopicAssignmentsForSkillAsync(
         this.skillId).then((response: AssignedSkill[]) => {
         this.topicsAssignments = {};
-        response.map((topic) => {
-          this.topicsAssignments[topic.topicName] = {
-            subtopicId: topic.subtopicId,
-            topicVersion: topic.topicVersion,
-            topicId: topic.topicId,
-          };
-        });
-        this.topicsAssignmentsAreFetched = true;
+        let allTopicIds = [];
+        for (let topic of response) {
+          allTopicIds.push(topic.topicId);
+        }
+
+        let topicIdsNotEligibleForUnassignment = [];
+        this.topicsAndSkillsDashboardBackendApiService
+          .fetchTopicIdToDiagnosticTestSkillIdsAsync(allTopicIds).then(
+            (dict) => {
+              for (let topicId in dict.topicIdToDiagnosticTestSkillIds) {
+                let diagnosticTestSkillIds = (
+                  dict.topicIdToDiagnosticTestSkillIds[topicId]);
+                if (diagnosticTestSkillIds.length === 1 &&
+                      diagnosticTestSkillIds.indexOf(this.skillId) !== -1) {
+                  topicIdsNotEligibleForUnassignment.push(topicId);
+                }
+              }
+
+              response = response.filter((topic) => {
+                return !(
```

Comment:
Why not drop both '!'s? A double-negative is harder to parse.

Also if possible I suggest reframing the variable to topicIdsEligibleForUnassignment. In general try to avoid lots of "not" constructs in the same thing -- it makes it harder to reason about.

Reply:
Updated